pkg/kubelet/node_manager.go

```diff
@@ -507,9 +509,11 @@ func (nm *realNodeManager) tryUpdateNodeStatus() error {
 	if node == nil {
 		return fmt.Errorf("no node instance returned for %q", nm.nodeName)
 	}
-	nm.lock.Lock()
-	defer nm.lock.Unlock()
-	nm.podCIDR = node.Spec.PodCIDR
+	if nm.reconcileCIDR {
+		nm.lock.Lock()
+		nm.podCIDR = node.Spec.PodCIDR
+		nm.lock.Unlock()
```

Comment:
This lock was previously held during the setNodeStatus & UpdateStatus calls (but I'm not sure whether that was necessary or just a side effect using defer). @cjcullen?